src/main/kotlin/com/openlattice/shuttle/IntegrationService.kt

```diff
@@ -0,0 +1,349 @@
+    )
+    private val authAPI = AuthAPI(
+            missionParameters.auth.getProperty("apiDomain"),
+            missionParameters.auth.getProperty("apiAudience"),
+            missionParameters.auth.getProperty("apiSecret")
+    )
+
+    companion object {
+        @JvmStatic
+        fun init(blackbox: Blackbox, dataModelService: EdmManager) {
+            if (blackbox.enabled) {
+                logEntityType = dataModelService.getEntityType(FullQualifiedName(blackbox.entityTypeFqn))
+                httpClient = OkHttpClient()
+            }
+        }
+    }
+
+    init {
+        executor.execute {
+            semaphore.acquire()
+
+            //load any jobs that were in progress or queued
+            integrationJobs.keySet(statusPredicate).forEach {
+                jobQueue.put(it)
+            }
+
+            while (true) {
+                if (semaphore.availablePermits() > 0) {
+                    val jobId = jobQueue.take()
+                    try {
+                        loadCargo(jobId)
+                    } catch (ex: Exception) {
+                        logger.info("Encountered exception $ex when trying to start integration job with id $jobId")
+                        jobQueue.add(jobId)
+                        logger.info("Integration job with id $jobId has been requeued")
+                    }
+                }
+            }
+        }
+    }
+
+    fun enqueueIntegrationJob(integrationName: String, integrationKey: UUID): UUID {
+        checkIntegrationExists(integrationName)
+        val integration = integrations.getValue(integrationName)
+        checkState(integrationKey == integration.key, "Integration key $integrationKey is incorrect")
+        val jobId = generateIntegrationJobId()
+        jobQueue.put(jobId)
+        integrationJobs[jobId] = IntegrationJob(integrationName, IntegrationStatus.QUEUED)
+        return jobId
+    }
+
+    private final fun loadCargo(jobId: UUID) {
+        val integrationJob = integrationJobs.getValue(jobId)
+        val integration = integrations.getValue(integrationJob.integrationName)
+        val token = getIdToken(missionParameters.auth)
+        val apiClient = ApiClient(integration.environment) { token }
+        val dataIntegrationApi = apiClient.dataIntegrationApi
+
+        //an integration object is expected to have a non-empty logEntitySetId,
+        //a non-null flight, and non-null key in order for the integration to run successfully
+        val flightPlan = mutableMapOf<Flight, Payload>()
+        val tableColsToPrint = mutableMapOf<Flight, List<String>>()
+        integration.flightPlanParameters.values.forEach {
+            val srcDataSourceProperties = Properties()
+            srcDataSourceProperties.putAll(it.source)
+            val srcDataSource = getSrcDataSource(srcDataSourceProperties)
+            val payload = JdbcPayload(readRateLimit.toDouble(), srcDataSource, it.sql, fetchSize, readRateLimit != 0)
```

Comment:
can you use a named parameter for this last one? it's a little unclear just reading this what parameter `readRateLimit != 0` might map to